src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -46,18 +50,90 @@ protected virtual CompletionItemRules GetCompletionItemRules(IReadOnlyList<ISymb
         private ImmutableArray<CompletionItem> CreateItems(
             ImmutableArray<ISymbol> symbols,
             SyntaxContext context,
-            bool preselect)
+            bool preselect,
+            ImmutableArray<ITypeSymbol> inferredTypes)
+        {
+            if (IsTargetTypeCompletionFilterExperimentEnabled(context.Workspace))
+            {
+                var symbolGroups = from symbol in symbols
+                                   let texts = GetDisplayAndSuffixAndInsertionText(symbol, context)
+                                   group symbol by texts into g
+                                   select g;
+
+                var itemListBuilder = ImmutableArray.CreateBuilder<CompletionItem>();
+
+                foreach (var symbolGroup in symbolGroups)
+                {
+                    var item = this.CreateItem(
+                            symbolGroup.Key.displayText, symbolGroup.Key.suffix, symbolGroup.Key.insertionText, symbolGroup.ToList(), context,
+                            invalidProjectMap: null, totalProjects: null, preselect: preselect);
+
+                    if (symbolGroup.Any(s => ShouldIncludeInTargetTypedCompletionList(s, inferredTypes, context.SemanticModel, context.Position)))
+                    {
+                        item = item.AddTag(WellKnownTags.TargetTypeMatch);
+                    }
+
+                    itemListBuilder.Add(item);
+                }
+
+                return itemListBuilder.ToImmutable();
+            }
+            else
+            {
+                var q = from symbol in symbols
+                        let texts = GetDisplayAndSuffixAndInsertionText(symbol, context)
+                        group symbol by texts into g
+                        select this.CreateItem(
+                            g.Key.displayText, g.Key.suffix, g.Key.insertionText, g.ToList(), context,
+                            invalidProjectMap: null, totalProjects: null, preselect: preselect);
+
+                return q.ToImmutableArray();
```

Comment:
This code is going to be removed after completion of the experiment. Correct?

If yes, then it is fine.
If not, we should reuse code between the if-else forks.

Reply:
Yes, it will be removed after the experiment. I wanted to leave the existing code _exactly_ as-is, so I didn't even attempt refactorings there.